jwst/associations/lib/rules_level2_base.py

```diff
@@ -894,6 +895,21 @@ def __init__(self):
             sources=['targetid'],
         )
 
+class Constraint_ExtCal(Constraint):
+    """Remove any nis_extcals from the associations, they
+       are NOT to be processed in the pipline!"""
+
+    def __init__(self):
+        super(Constraint_ExtCal, self).__init__(
+            [
+                DMSAttrConstraint(
+                    name='exp_type',
+                    sources=['exp_type'],
+                    value='nis_extcal'
```

Comment:
Not sure what the EXP_TYPE values look like in the pool files, from which these entries are read. Are the values in the pool files always uppercase (e.g. "NIS_EXTCAL") and, if so, does that make a difference for this matching?

Reply:
Yes, the pool files have all of these values in upper case. It does not make a difference since they are all converted to lower case in the association code. Since I was looking at (copying from) the pool file I used upper case as written.